show/main.py

```diff
@@ -768,23 +771,241 @@ def get_bgp_peer():
         bgp_peer.setdefault(local_addr, [neighbor_name, neighbor_ip])
     return bgp_peer
 
+def print_ip_routes(route_info):
+    """
+    Sample Entry output
+    B>*172.16.8.2/32 [20/0] via 10.0.0.47, Ethernet92, 03:33:01
+    B>*192.168.114.96/32 [20/0] via 10.0.0.1, PortChannel0002, 03:30:39
+      *                         via 10.0.0.5, PortChannel0005, 03:30:39
+      *                         via 10.0.0.9, PortChannel0008, 03:30:39
+      *                         via 10.0.0.13, PortChannel0011, 03:30:39
+      *                         via 10.0.0.17, PortChannel0014, 03:30:39
+      *                         via 10.0.0.21, PortChannel0017, 03:30:39
+      *                         via 10.0.0.25, PortChannel0020, 03:30:39
+      *                         via 10.0.0.29, PortChannel0023, 03:30:39
+    B 10.0.107.0/31 [200/0] via 10.0.107.1, inactive 00:10:15
+    K>*0.0.0.0/0 [0/0] via 10.3.146.1, eth0, 00:25:22
+    B 0.0.0.0/0 [20/0] via 10.0.0.1, PortChannel0002, 03:31:52
+                      via 10.0.0.5, PortChannel0005, 03:31:52
+                      via 10.0.0.9, PortChannel0008, 03:31:52
+                      via 10.0.0.13, PortChannel0011, 03:31:52
+                      via 10.0.0.17, PortChannel0014, 03:31:52
+                      via 10.0.0.21, PortChannel0017, 03:31:52
+                      via 10.0.0.25, PortChannel0020, 03:31:52
+                      via 10.0.0.29, PortChannel0023, 03:31:52
+    S 0.0.0.0/0 [200/0] via 10.3.146.1, eth0, 03:35:18
+    C>*10.0.0.62/31 is directly connected, Ethernet124, 03:34:00
+    """
+    # This method is following what zebra_vty.c does when handling the route parsing printing
+    # The route_info is a json file which is treated as a dictionary of a bunch of route + info
+    # This interpretation is based on FRR 7.2 branch. If we later moved on to a new branch, we may
+    # have to rexamine if there are any changes made that may impact the parsing logic
+    proto_code = {"system":'X', "kernel":'K', "connected":'C', "static":'S', "rip":'R', "ripng":'R',
+            "ospf":'O', "ospf6":'O', "isis":'I', "bgp":'B', "pim":'P', "hsls":'H', "olsr":'o', "babel":'A'}
+    mpls_label_code = { 0:"IPv4 Explicit Null", 1:"Router Alert", 2:"IPv6 Explicit Null", 3:"implicit-null", 4:"Reserved (4)",
+            5:"Reserved (5)", 6:"Reserved (6)", 7:"Entropy Label Indicator", 8:"Reserved (8)", 9:"Reserved (9)", 10:"Reserved (10)",
+            11:"Reserved (11)", 12:"Reserved (12)", 13:"Generic Associated Channel", 14:"OAM Alert", 15:"Extension"}
```

Comment:
Let's try to maintain 79 chars in a line if possible.

Reply:
This is debatable now that we all have wider screens. I'm OK with making one exception to the PEP8 standard and extending the max line length to around 100 chars. We can discuss this more offline. Using a tool like `autopep8 --max-line-length 100` can be helpful.

Reply:
Sure, it's not a hardcore requirement. With 80 chars I get more vertical splits in editor :).

Reply:
Sure I will reduce the length shorter to not exceed more than 100 chars.
The code can become harder to read if need to strictly follow this 79 chars limitation.